app/models/location.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Value object - date range
+class Location
+  attr_reader :state, :street_location, :city, :zipcode
+
+  def initialize(state:, street_location: nil, city: nil, zipcode: nil)
+    @state = state
+    @street_location = street_location
+    @city = city
+    @zipcode = zipcode
+  end
+
+  def to_s
+    [street_location, city, state, ("Zipcode: #{zipcode}" if zipcode)].compact.join(", ")
```

Comment:
Why the logic for zipcode?

Reply:
@rlgreen91  I could directly append. Only reason I went with this approach after going through some address format in internet. But other thing I observed is, address format looks different with different country. Let's discuss about the format and then I will make changes accordingly.

Reply:
@rlgreen91  I have removed the logic of zipcode for now. Let me know if we change format for address?